octree/include/pcl/octree/impl/octree_pointcloud.hpp

```diff
@@ -66,7 +64,11 @@ pcl::octree::OctreePointCloud<PointT, LeafContainerT, BranchContainerT, OctreeT>
 , bounding_box_defined_(false)
 , max_objs_per_leaf_(0)
 {
-  assert(resolution > 0.0f);
+  if (resolution <= 0.0) {
```

Comment:
While it makes sense to not use assert here, I would either set the resolution to 1 (or some other value), or throw an exception